mpc/types/ethtx.go

```diff
@@ -0,0 +1,167 @@
+package types
```

Comment:
Can you explain why we need this?

Reply:
There are some differences in RLP decoding in go-ethereum 1.14 and 1.10. The binary data of unsigned transaction does not have r, s and v encoded, if we use `tx.UnmarshalBinary` provided in go-ethereum 1.14, it will cause an error (too many elements) when decoding. So I copied over the data structures from go-ethereum and set the r, s, v to optional to avoid this error.

Reply:
I don't think so.

Please give an example for it.

Reply:
1. Create a dynamic fee tx
<img width="475" alt="image" src="https://github.com/user-attachments/assets/6c51de35-b2c1-4cb3-b14c-257eaaf4fb93">

2. Decode the transaction with `tx.UnmarshalBinary`
<img width="1434" alt="image" src="https://github.com/user-attachments/assets/26a7c862-6216-4e72-872f-bf97127a4d8f">
<img width="682" alt="image" src="https://github.com/user-attachments/assets/c4c99ec3-de5c-44be-abb8-3e8e627f309a">

Also that one thing to notice, I played with different transaction types, it seems that this error will only happens on typed transactions (access list / dynamic fee / blob), not on legacy tx.

Reply:
Same tx data with the optional decoding:
<img width="1440" alt="image" src="https://github.com/user-attachments/assets/bfd9388a-4adb-45b6-bfc1-a9318e4b8867">
<img width="1454" alt="image" src="https://github.com/user-attachments/assets/fee68767-61fb-4b93-8c5d-65d7e5fc5066">

Reply:
I guess this might be the reason? But I'm not sure the relation between gencodec tag and RLP decode.
 
<img width="681" alt="image" src="https://github.com/user-attachments/assets/c4c518f7-58bf-463b-95b8-96a50cfe03ff">

Reply:
I just checked the code of the ether.js, it doesn't add the sig fields to the rlp bytes

https://github.com/ethers-io/ethers.js/blob/9e7e7f3e2f2d51019aaa782e6290e079c38332fb/src.ts/transaction/transaction.ts#L386-L389

But it should work for geth

```
package main

import (
	"math/big"

	"github.com/ethereum/go-ethereum/common"
	"github.com/ethereum/go-ethereum/core/types"
)

func main() {
	var a common.Address
	tx1 := types.NewTx(&types.DynamicFeeTx{
		ChainID:   big.NewInt(1),
		GasTipCap: big.NewInt(1e9),
		GasFeeCap: big.NewInt(1e9),
		Gas:       1e5,
		To:        &a,
		Value:     big.NewInt(0),
	})

	tx1Raw, err := tx1.MarshalBinary()
	if err != nil {
		panic(err)
	}

	var tx2 types.Transaction
	if err := tx2.UnmarshalBinary(tx1Raw); err != nil {
		panic(err)
	}
}
```

Fair enough, we can accept the change.